### PR TITLE
[SPARK-35588][PYTHON][DOCS] Merge Binder integration and quickstart notebook for pandas API on Spark

### PR DESCRIPTION
**Pull request by HyukjinKwon**
### What changes were proposed in this pull request?

This PR proposes to fix:
- the Binder integration of pandas API on Spark, and merge them together with the existing PySpark one.
- update quickstart of pandas API on Spark, and make it working

The notebooks can be easily reviewed here:

https://mybinder.org/v2/gh/HyukjinKwon/spark/SPARK-35588-3?filepath=python%2Fdocs%2Fsource%2Fgetting_started%2Fquickstart_ps.ipynb

Original page in Koalas: https://koalas.readthedocs.io/en/latest/getting_started/10min.html

### Why are the changes needed?

- To show the working examples of quickstart to end users.
- To allow users to try out the examples without installation easily.

### Does this PR introduce _any_ user-facing change?

No to end users because the existing quickstart of pandas API on Spark is not released yet.

### How was this patch tested?

I manually tested it by uploading built Spark distribution to Binder. See https://github.com/HyukjinKwon/spark/commit/3bc15310a05b5005aca8e753e2853a076a5d4e68

### CODE REVIEW

**Comment by HyukjinKwon** [comment]
cc @ueshin @xinrong-databricks @itholic FYI

**Comment by HyukjinKwon** [comment]
BTW, the plotly charts in the notebooks make too many diffs .. like 14K diff .. next time, we should maybe make the output deterministic so the diff is small. Currently. there are some examples using randoms which make the chart different.

File: python/docs/source/getting_started/quickstart_df.ipynb [hunk 1]
```diff
@@ -1167,7 +1167,7 @@
    "name": "python",
    "nbconvert_exporter": "python",
    "pygments_lexer": "ipython3",
-   "version": "3.7.8"
+   "version": "3.7.10"
```
Comment (xinrong-databricks) [inline python/docs/source/getting_started/quickstart_df.ipynb:1170]: Out of curiosity, why do we upgrade the `version`?

**Review by xinrong-meng — APPROVED** [review]

**Comment by xinrong-databricks** [comment]
LGTM, thanks!

File: python/docs/source/getting_started/quickstart_df.ipynb (hunk 1 above)
Comment (HyukjinKwon) [reply python/docs/source/getting_started/quickstart_df.ipynb:1170]: Seems like Binder upgraded their Python version .. I executed this notebook after my change to make sure it's working :-) ..

Comment (xinrong-databricks) [reply python/docs/source/getting_started/quickstart_df.ipynb:1170]: Got it, thanks! :)

**Comment by HyukjinKwon** [comment]
Merged to master.

The tests already passed. I just made some minor changes on the wording a lot by self-reviewing. I double checked that in my local too

**Pull request closed without merging**